sequencer/src/api/sql.rs

```diff
@@ -294,27 +294,40 @@ impl CatchupStorage for SqlStorage {
             return Ok(Vec::new());
         }
 
-        // get the latest balance for each account
-        // We use ROW_NUMBER() to rank entries by created height per account,
-        // then select only the most recent entry (rn = 1) for each account.
-        let rows = query_as::<(Value, Value)>(&format!(
+        // get the latest balance for each account.
+        // use DISTINCT ON for Postgres
+        // use ROW_NUMBER() as DISTINCT ON is not supported for SQLite
+        #[cfg(not(feature = "embedded-db"))]
+        let query = format!(
+            "SELECT DISTINCT ON (idx) idx, entry
+               FROM {}
+              WHERE idx IS NOT NULL AND created <= $1
+              ORDER BY idx DESC, created DESC
+              LIMIT $2 OFFSET $3",
+            RewardMerkleTreeV2::state_type()
+        );
+
+        #[cfg(feature = "embedded-db")]
+        let query = format!(
             "SELECT idx, entry FROM (
                  SELECT idx, entry, ROW_NUMBER() OVER (PARTITION BY idx ORDER BY created DESC) as \
              rn
                    FROM {}
-                  WHERE created <= $1 AND idx IS NOT NULL AND entry IS NOT NULL
+                  WHERE created <= $1 AND idx IS NOT NULL
              ) sub
              WHERE rn = 1
              ORDER BY idx
              LIMIT $2 OFFSET $3",
             RewardMerkleTreeV2::state_type()
-        ))
-        .bind(height as i64)
-        .bind(limit as i64)
-        .bind(offset as i64)
-        .fetch_all(tx.as_mut())
-        .await
-        .context("loading reward accounts from storage")?;
+        );
```

Comment:
I wonder if we want to just not support this endpoint at all when using sqlite, since we know this doesn't perform well. Like just return `Err("not supported")` here

Reply:
yeah, but if we enable sqlite with pruning settings then this query should be fine I think

Reply:
Oh yeah good point